playground/examples.js

```diff
@@ -54,7 +54,7 @@ function jsonExample() {
     const Comma = createToken({name: "Comma", pattern: /,/});
     const Colon = createToken({name: "Colon", pattern: /:/});
     const StringLiteral = createToken({
-        name: "StringLiteral", pattern: /"(:?[^\\"\n\r]+|\\(:?[bfnrtv"\\/]|u[0-9a-fA-F]{4}))*"/
+        name: "StringLiteral", pattern: /"(?:[^\\"]|\\(?:[bfnrtv"\\/]|u[0-9a-fA-F]{4}))*"/
```

Comment:
I recall making this mistake with the syntax of the none capturing group.
I must not have fixed it everywhere.
However I think that disallowing the newlines chars should be kept, see snippet below:

```javascript
const res = /"(?:[^\\"]|\\(?:[bfnrtv"\\/]|u[0-9a-fA-F]{4}))*"/.test(`"


"`)
// res is true, but that is not a valid JSON string.
```

Reply:
Yeah you're right, I didn't take the ^ into account and thought it was allowing newlines.